src/bucket.ts

```diff
@@ -989,6 +1004,140 @@ class Bucket extends ServiceObject {
     this.getFilesStream = paginator.streamify('getFiles');
   }
 
+  /**
+   * Set the object lifecycle rules for objects in this bucket.
+   *
+   * By default, the object lifecycle rule you provide to this method will be
+   * appended to any existing ones. To replace all existing rules, supply the
+   * `options` argument, setting `append` to `false`.
+   *
+   * @see [Object Lifecycle Management]{@link https://cloud.google.com/storage/docs/lifecycle}
+   * @see [Buckets: patch API Documentation]{@link https://cloud.google.com/storage/docs/json_api/v1/buckets/patch}
+   *
+   * @param {LifecycleRule} rule The new lifecycle rule to be added to objects
+   *     in this bucket.
+   * @param {AddLifecycleRuleOptions} [options] Configuration object.
+   * @param {SetBucketMetadataCallback} [callback] Callback function.
+   * @returns {Promise<SetBucketMetadataResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('albums');
+   *
+   * //-
```

Comment:
Please add examples of how to delete all the rules in a single request and iterate through them.